pkg/engine2/solution_context/resource_configuration.go

```diff
@@ -71,6 +80,25 @@ func ConfigureResource(
 	return nil
 }
 
+func AddDeploymentDependenciesFromVal(
+	ctx SolutionContext,
+	resource *construct.Resource,
+	val any,
+) error {
+	var errs error
+	ids := getResourcesFromValue(val)
+	for _, id := range ids {
+		if resource.ID.Matches(id) {
```

Comment:
Can this happen? How would this look like in IaC?

Reply:
yeah for the uri on the lb we use it as a ref to itself so that it can get the integrations route

Reply:
👍 right, for the deployment variables, those are substituted